packages/salesforcedx-vscode-apex/src/views/testOutline.ts

```diff
@@ -0,0 +1,666 @@
+const DARK_RED_BUTTON = ospath.join(
+  __filename,
+  '..',
+  '..',
+  '..',
+  '..',
+  'resources',
+  'dark',
+  'testFail.svg'
+);
+const DARK_GREEN_BUTTON = ospath.join(
+  __filename,
+  '..',
+  '..',
+  '..',
+  '..',
+  'resources',
+  'dark',
+  'testPass.svg'
+);
+
+export type FullTestResult = {
+  summary: TestSummary;
+  tests: TestResult[];
+};
+
+export type TestSummary = {
+  outcome: string;
+  testsRan: number;
+  passing: number;
+  failing: number;
+  skipped: number;
+  passRate: string;
+  failRate: string;
+  testStartTime: string;
+  testExecutionTime: string;
+  testTotalTime: string;
+  commandTime: string;
+  hostname: string;
+  orgId: string;
+  username: string;
+  testRunId: string;
+  userId: string;
+};
+
+export type TestResult = {
+  ApexClass: ApexClass;
+  MethodName: string;
+  Outcome: string;
+  RunTime: number;
+  Message: string;
+  StackTrace: string;
+  FullName: string;
+};
+
+export type ApexClass = {
+  attributes: { type: string };
+  Id: string;
+  Name: string;
+  NamespacPrefix: string;
+};
+
+export type ApexGroupInfo = {
+  outcome: string;
+  summary: TestSummary;
+};
+
+export class ApexTestOutlineProvider implements vscode.TreeDataProvider<Test> {
+  private _onDidChangeTreeData: vscode.EventEmitter<
+    Test | undefined
+  > = new vscode.EventEmitter<Test | undefined>();
+  public readonly onDidChangeTreeData: vscode.Event<Test | undefined> = this
+    ._onDidChangeTreeData.event;
+
+  private eventsEmitter = new events.EventEmitter();
+
+  private apexTestMap: Map<string, Test> = new Map<string, Test>();
+  private head: Test | null;
+  private static testStrings: Set<string> = new Set<string>();
+
+  constructor(
+    private path: string,
+    private apexClasses: vscode.Uri[],
+    private apexTestInfo: ApexTestRequestInfo[] | null
+  ) {
+    this.head = null;
+    this.getAllApexTests(this.path);
+    // Now, activate events that we need
+    this.eventsEmitter.on('Delete Folder', this.deleteFolderRecursive); // Activate event to delete folder
+    this.eventsEmitter.on('Show Highlight', this.updateSelection);
+  }
+
+  public getHead(): Test {
+    if (this.head === null) {
+      return this.getAllApexTests(this.path);
+    }
+    return this.head;
+  }
+
+  public getChildren(element: Test): Thenable<Test[]> {
+    if (element) {
+      return Promise.resolve(element.children);
+    } else {
+      if (this.head && this.head.children.length > 0) {
+        return Promise.resolve(this.head.children);
+      } else {
+        const emptyArray = new Array<ApexTest>();
+        emptyArray.push(new ApexTest('No tests in folder', null, 0));
+        return Promise.resolve(emptyArray);
+      }
+    }
+  }
+
+  public getTreeItem(element: Test): vscode.TreeItem {
+    if (element) {
+      return element;
+    } else {
+      this.getAllApexTests(this.path);
+      if (!(this.head && this.head.children.length > 0)) {
+        this.head = new ApexTest('No tests in folder', null, 0);
```

Comment:
@jkalloor0701 - All messages should be externalized.
@ruthemmanuelle - Comments?

Reply:
@jkalloor0701 - Is this the same situation as the one above?

Reply:
Yes, it is